lib/nrf_modem_lib/sanity.c

```diff
@@ -91,3 +98,66 @@ BUILD_ASSERT(ESHUTDOWN       == NRF_ESHUTDOWN,       "Errno not aligned with nrf
 BUILD_ASSERT(EHOSTDOWN       == NRF_EHOSTDOWN,       "Errno not aligned with nrf_errno.");
 BUILD_ASSERT(ESOCKTNOSUPPORT == NRF_ESOCKTNOSUPPORT, "Errno not aligned with nrf_errno.");
 BUILD_ASSERT(ETOOMANYREFS    == NRF_ETOOMANYREFS,    "Errno not aligned with nrf_errno.");
+
+/* Shared memory sanity check */
+
+#define SRAM_BASE 0x20000000
+#define SHMEM_RANGE KB(128)
+#define SHMEM_END (SRAM_BASE + SHMEM_RANGE)
+
+#define SHMEM_IN_USE \
+	(PM_NRF_MODEM_LIB_SRAM_ADDRESS + PM_NRF_MODEM_LIB_SRAM_SIZE)
+
+BUILD_ASSERT(SHMEM_IN_USE < SHMEM_END,
+	     "The libmodem shmem configuration exceeds the range of "
+	     "memory readable by the Modem core");
+
+/* Socket values sanity check */
+
+BUILD_ASSERT(AF_UNSPEC == NRF_AF_UNSPEC, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(AF_INET == NRF_AF_INET, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(AF_INET6 == NRF_AF_INET6, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(AF_PACKET == NRF_AF_PACKET, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(SOCK_STREAM == NRF_SOCK_STREAM, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOCK_DGRAM == NRF_SOCK_DGRAM, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOCK_RAW == NRF_SOCK_RAW, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(IPPROTO_IP == NRF_IPPROTO_IP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_TCP == NRF_IPPROTO_TCP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_UDP == NRF_IPPROTO_UDP, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_IPV6 == NRF_IPPROTO_IPV6, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(IPPROTO_RAW == NRF_IPPROTO_RAW, "Socket value not aligned with modemlib.");
+/* Comparison is inverted here to prevent `WARNING:CONSTANT_COMPARISON` from checkpatch */
+BUILD_ASSERT(NRF_SPROTO_TLS1v2 == IPPROTO_TLS_1_2, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(NRF_SPROTO_DTLS1v2 == IPPROTO_DTLS_1_2, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(SOL_TLS == NRF_SOL_SECURE, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(SOL_SOCKET == NRF_SOL_SOCKET, "Socket value not aligned with modemlib.");
+
+BUILD_ASSERT(MSG_DONTWAIT == NRF_MSG_DONTWAIT, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(MSG_PEEK == NRF_MSG_PEEK, "Socket value not aligned with modemlib.");
+BUILD_ASSERT(MSG_WAITALL == NRF_MSG_WAITALL, "Socket value not aligned with modemlib.");
```

Comment:
This seems to be a bit messy upstream, but w/o POSIX APIs enabled, those symbols would not be defined (need to prefix them with `ZSOCK_`). This causes twister failures in a few projects.

Reply:
@rlubos Yes, I've been looking into this for the past hour